clients/itwin/src/Client.ts

```diff
@@ -83,10 +89,17 @@ export abstract class Client {
       return this._url;
     }
 
-    const urlDiscoveryClient: UrlDiscoveryClient = new UrlDiscoveryClient();
+    const resolvedRegion = Config.App.getNumber(UrlDiscoveryClient.configResolveUrlUsingRegion, 0);
     const searchKey: string = this.getUrlSearchKey();
+    const configuredUrl: string = Config.App.query(`imjs_url_${resolvedRegion}_${searchKey}`) ?? this.baseUrl;

```

Comment:
What do you think of having a single config variable here that could append `dev-` or `qa-` in front of `this.baseUrl` if it's defined?  Then there will be two optional ways to customize the url for each client before the UrlDiscoveryClient is invoked;

1. Set `this.baseUrl` in the implementation of the client
2. Add a `imjs_url_prefix` variable that will be appeded to every url when set.

It does make it a bit more difficult to point a single client at a different url than all the others but I don't see many use-cases for that and could just lead to confusion.  Plus if that's desired the baseUrl can be changed at the top-level.  Thinking a bit more long-term, the structure of this needs to change more drastically to prepare for fully moving to the iTwin Platform APIs along with supporting the iTwin Stack moving forward.  Not something we need to tackle in this PR though.

@ramanujam-raman

Reply:
I think it would introduce unnecessary code for parsing and gluing URLs back together. I have changes for imodeljs-config that set all different environment URLs -- that should be sufficient and less complex.

Reply:
I second Caleb's suggestion - it would be good to minimize dependency on the env file - that makes this more maintainable and also enforces consistency in naming of the URLs. Just use a simple prefix to prepend to every URL when set. 

I see this as a simpler option - also, the caching of the URL keeps the processing to a minimum - I don't see much of a cost to this.

Reply:
Added code to prepend hostname with a configured prefix when this.baseUrl is set

Reply:
Okay, I'm wondering if we want to have both ways to configure for each environment since there already is a way to configure the relying party.

Any objection to removing the per-environment support of the relying party?

Reply:
Removed checks for per-environment configuration